packages/auto-utils/src/api.ts

```diff
@@ -1,29 +1,40 @@
 import { ApiPromise, WsProvider } from '@polkadot/api'
 import { getNetworkDomainRpcUrls, getNetworkRpcUrls } from './network'
-import type { DomainInput, NetworkInput } from './types/network'
+import type { ActivateInput, ApiOptions, DomainInput, NetworkInput } from './types/network'
 
-export const activate = async (input?: NetworkInput): Promise<ApiPromise> => {
-  // Get the first rpc urls for the network
-  const rpcUrl = getNetworkRpcUrls(input)
+export const createConnection = async (
+  endpoint: string,
+  options?: ApiOptions,
+): Promise<ApiPromise> => {
   // Create the provider
-  const provider = new WsProvider(rpcUrl[0])
+  const provider = new WsProvider(endpoint)
   // Create the API instance
-  const api = await ApiPromise.create({ provider })
+  const api = await ApiPromise.create({
+    ...options,
+    noInitWarn: options?.noInitWarn ?? true,
+    provider,
+  })
   await api.isReady
 
   return api
 }
 
-export const activateDomain = async (input: DomainInput): Promise<ApiPromise> => {
+export const activate = async (input?: ActivateInput<NetworkInput>): Promise<ApiPromise> => {
```

Comment:
It is not very clear to me what `input` represents? Is it options or params? The naming feels off here, but I may just not be following.

Reply:
yeah I could replace there by options or params...

it take in a optional networkId (default to 3h if nothing is pass) and a optional api, so if you activate multiple wallet you can initialize 1 api and use the same for all wallet to save time